sdk/deviceprovisioningservices/Microsoft.Azure.Management.DeviceProvisioningServices/tests/ScenarioTests/DeviceProvisioningClientTest.cs

```diff
@@ -1,153 +1,164 @@
+            while (attempts > 0)
             {
-                this.Initialize(context);
-                string testName = "unitTestingDPSUpdateSku";
-                var resourceGroup = this.GetResourceGroup(testName);
-                var service = this.GetService(testName, testName);
-                //update capacity
-                service.Sku.Capacity += 1;
-                var attempts = Constants.ArmAttemptLimit;
-                var success = false;
-                while (attempts > 0 && !success)
+                try
                 {
-                    try
-                    {
-                        var updatedInstance =
-                   this.provisioningClient.IotDpsResource.CreateOrUpdate(resourceGroup.Name, service.Name,
-                       service);
-                        Assert.Equal(service.Sku.Capacity, updatedInstance.Sku.Capacity);
-                        success = true;
-                    }
-                    catch
-                    {
-                        //Let ARM finish
-                        System.Threading.Thread.Sleep(Constants.ArmAttemptWaitMS);
-                        attempts--;
-                    }
+                    ProvisioningServiceDescription updatedInstance = await _provisioningClient
+                        .IotDpsResource.CreateOrUpdateAsync(
+                            rg.Name,
+                            service.Name,
+                            service)
+                        .ConfigureAwait(false);
+                    updatedInstance.Sku.Capacity.Should().Be(service.Sku.Capacity);
+                    break;
+                }
+                catch
+                {
+                    // Let ARM finish
+                    await Task.Delay(Constants.ArmAttemptWaitMs).ConfigureAwait(false);
+                    attempts--;
                 }
-
-
-
             }
         }
-        [Fact(Skip = "Needs re-recording")]
-        //[Fact]
-        public void CreateFailure()
-        {
-            using (var context = MockContext.Start(this.GetType()))
-            {
-
-                this.Initialize(context);
-                var testName = "unitTestingDPSCreateUpdateInvalidName";
-                this.GetResourceGroup(testName);
-
-
-                //try to create a DPS service
-                var createServiceDescription = new ProvisioningServiceDescription(Constants.DefaultLocation,
-                    new IotDpsPropertiesDescription(),
-                    new IotDpsSkuInfo(Constants.DefaultSku.Name,
-                        Constants.DefaultSku.Tier,
-                        Constants.DefaultSku.Capacity
-                    ));
 
-                var badCall = new Func<ProvisioningServiceDescription>(() => this.provisioningClient.IotDpsResource.CreateOrUpdate(
-                    testName,
-                    $"1�1{testName}!!!", //We dont't allow most punctuation, leading numbers, etc
+        [Fact]
+        public async Task CreateFailure()
```

Comment:
This test fails on Linux and Mac because the expected exception type is different.

> Expected a <Microsoft.Azure.Management.DeviceProvisioningServices.Models.ErrorDetailsException> to be thrown, but found <System.Collections.Generic.

Any idea why that is, @allenjzhang ?

Reply:
Probably test case needs to be updated. The `found` got cut off in comment but most likely succeeded and returned a collection of result instead of throwing exceptions.

Reply:
They ran (live and playback) fine locally, and passed for Windows.

Reply:
Sorry, I updated my note. That was the output from the pipeline.

It expected one exception (which is thrown in Windows), but got a different one in Mac and Linux.

Reply:
Do you have the call stack? Most time when issues either build or test fails only on Linux, it may be due to casing.

Reply:
https://dev.azure.com/azure-sdk/public/_build/results?buildId=789123&view=ms.vss-test-web.build-test-results-tab&runId=17426864&resultId=100003&paneView=debug

Reply:
Error log of this test. The error shows that the matching recording file could not be found. This is due to REST API signature changes. You should rerecord.
https://dev.azure.com/azure-sdk/public/_build/results?buildId=789123&view=logs&j=ad07b744-5a1a-5d54-7693-fcb02e666633&t=88540a4e-0b51-541f-e623-7b2636eab300&l=27459

Reply:
Looks to be the invalid instance name used in this failure test, and how it is escaped on each platorm.

Reply:
I re-recorded it twice. You can see the new json file in the PR. I'm trying different input for the name for now.

Reply:
My input change passed on Linux.